src/fstapp_server.erl

```diff
@@ -1,64 +1,96 @@
 -module(fstapp_server).
 -behaviour(gen_server).
 
--export([start_link/0, start_get_metrics/0, change_freq_metrics/1, stop_get_metrics/0]).
+-export([start_link/0, start_get_metrics/0, change_freq_metrics/1, stop_get_metrics/0,
+	 metrics/1]).
 
 -export([init/1, handle_call/3, handle_cast/2, handle_info/2,
 	terminate/2]).
 
+%% state
 -record(state, {timer_ref, freq}).
 
-%% ------
-%% API
-%% ------
+
+%%%====================================================================
+%% API functioncs
+%%=====================================================================
 
 start_link() ->
 	gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
 
+-spec start_get_metrics() -> term().
 start_get_metrics() ->
 	gen_server:call(?MODULE, start_get_metrics).
 
+-spec change_freq_metrics(number()) -> term().
 change_freq_metrics(Time) ->
 	gen_server:call(?MODULE, {change_freq_metrics, Time}).
 
+-spec stop_get_metrics() -> term().
 stop_get_metrics() ->
 	gen_server:call(?MODULE, stop_collecting_metrics).
 
-%% ------
-%% gen_server callbacks
-%% ------
+%%%====================================================================
+%% gen_server functioncs
+%%=====================================================================
 
+%% @hidden
 init([]) ->
 	TimerRef = erlang:send_after(5000, self(), collect_metrics),
 	{ok, #state{timer_ref=TimerRef, freq=5000}}.
 
+%% @hidden
 handle_call(start_get_metrics, _From, State) ->
 	erlang:cancel_timer(State#state.timer_ref),
 	TimerRef = erlang:send_after(State#state.freq, self(), collect_metrics),
-	Reply = ok,
 	{reply, ok, #state{timer_ref=TimerRef, freq=State#state.freq}};
 handle_call({change_freq_metrics, Time}, _From, State) ->
 	erlang:cancel_timer(State#state.timer_ref),
-	TimerRef = erlang:send_after(Time*1000, self(), collect_metrics),
+	erlang:send_after(Time*1000, self(), collect_metrics),
 	{reply, ok, #state{timer_ref=State#state.timer_ref, freq=Time*1000}};
```

Comment:
this state.timer_ref is already canceled. So this is not going to work. You must to use the new TimerRef created